packages/liveblocks-react/src/comments/CommentsRoom.ts

```diff
@@ -97,153 +99,239 @@ export type RoomThreads<TThreadMetadata extends BaseMetadata> =
       error?: never;
     };
 
+type ThreadsRequestInfo<TThreadMetadata extends BaseMetadata> = {
+  fetcher: Promise<ThreadData<TThreadMetadata>[]>;
+  timestamp: number;
+};
+
+type MutationInfo = {
+  startTime: number;
+  endTime: number;
+};
+
+function createThreadsManager<TThreadMetadata extends BaseMetadata>() {
+  let cache: RoomThreads<TThreadMetadata> | undefined; // Stores the current cache state (threads)
+  let request: ThreadsRequestInfo<TThreadMetadata> | undefined; // Stores the currently active revalidation request
+  let mutation: MutationInfo | undefined; // Stores the start and end time of the currently active mutation
+
+  const eventSource = makeEventSource<
+    RoomThreads<TThreadMetadata> | undefined
+  >();
+
```

Comment:
Updated to use `EventSource` API. One of the reasons why I opted to use `EventTarget` Web API was that I wanted to use standard Web APIs, so anyone reading the code in future won't have to learn a new API if they already knew the `EventTarget` API but I have to agree that this `EventSource` API provides a much better experience. Thanks for the suggestion @nvie